myriad-scheduler/src/main/java/org/apache/myriad/scheduler/ServiceTaskFactoryImpl.java

```diff
@@ -80,174 +68,33 @@ public TaskInfo createTask(Offer offer, FrameworkID frameworkId, TaskID taskId,
-    }
+    //taskBuilder.addResources(Resource.newBuilder().setName("ports").setType(Value.Type.RANGES).setRanges(valueRanger.build()));
     taskBuilder.setCommand(commandInfo);
     if (cfg.getContainerInfo().isPresent()) {
       taskBuilder.setContainer(taskUtils.getContainerInfo());
     }
     return taskBuilder.build();
   }
 
-  @VisibleForTesting
-  CommandInfo createCommandInfo(ServiceResourceProfile profile, String executorCmd) {
-    MyriadExecutorConfiguration myriadExecutorConfiguration = cfg.getMyriadExecutorConfiguration();
-    CommandInfo.Builder commandInfo = CommandInfo.newBuilder();
-    Map<String, String> envVars = cfg.getYarnEnvironment();
-    if (MapUtils.isNotEmpty(envVars)) {
-      Protos.Environment.Builder yarnHomeB = Protos.Environment.newBuilder();
-      for (Map.Entry<String, String> envEntry : envVars.entrySet()) {
-        Protos.Environment.Variable.Builder yarnEnvB = Protos.Environment.Variable.newBuilder();
-        yarnEnvB.setName(envEntry.getKey()).setValue(envEntry.getValue());
-        yarnHomeB.addVariables(yarnEnvB.build());
-      }
-      commandInfo.mergeEnvironment(yarnHomeB.build());
-    }
-
-    if (myriadExecutorConfiguration.getNodeManagerUri().isPresent()) {
-      //Both FrameworkUser and FrameworkSuperuser to get all of the directory permissions correct.
-      if (!minimumUserSet(cfg)) {
-        throw new RuntimeException("Trying to use remote distribution, but frameworkUser" + "and/or frameworkSuperUser not set!");
-      }
-
-      LOGGER.info("Using remote distribution");
-      String clGeneratedCommand = clGenerator.generateCommandLine(profile, null);
-
-      String nmURIString = myriadExecutorConfiguration.getNodeManagerUri().get();
-
-      //Concatenate all the subcommands
-      String cmd = clGeneratedCommand + " " + executorCmd;
-
-      //get the nodemanagerURI
-      //We're going to extract ourselves, so setExtract is false
-      LOGGER.info("Getting Hadoop distribution from:" + nmURIString);
-      URI nmUri = URI.newBuilder().setValue(nmURIString).setExtract(false).build();
-
-      //get configs directly from resource manager
-      String configUrlString = clGenerator.getConfigurationUrl();
-      LOGGER.info("Getting config from:" + configUrlString);
-      URI configUri = URI.newBuilder().setValue(configUrlString).build();
-
-      LOGGER.info("Slave will execute command:" + cmd);
-      commandInfo.addUris(nmUri).addUris(configUri).setValue("echo \"" + cmd + "\";" + cmd);
-      commandInfo.setUser(cfg.getFrameworkSuperUser().get());
-
-    } else {
-      commandInfo.setValue(executorCmd);
-    }
-    return commandInfo.build();
-  }
-
-  private Boolean minimumUserSet(MyriadConfiguration conf) {
-    return (cfg.getFrameworkUser().isPresent() || cfg.getFrameworkSuperUser().isPresent());
-  }
-  
   @Override
   public ExecutorInfo getExecutorInfoForSlave(FrameworkID frameworkId, Offer offer, CommandInfo commandInfo) {
```

Comment:
Would it make more sense to return an empty/default ExecutorInfo object? Is there a risk of an NPE elsewhere?